csharp/src/Apache.Arrow.Compression/Apache.Arrow.Compression.csproj

```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="CommunityToolkit.HighPerformance" Version="8.0.0" />
```

Comment:
This dependency is only needed for the  `AsStream` extension methods on `Memory<byte>` and `ReadOnlyMemory<byte>` which could be re-implemented if we wanted to minimize extra dependencies.

Reply:
I think this is fine.

Reply:
I found that the LZ4 library offers APIs that takes Spans. We should just use those APIs instead, and then we can get rid of this dependency.

https://github.com/MiloszKrajewski/K4os.Compression.LZ4/blob/fa8b8e038b500d565efe12769db097852a28ddf7/src/K4os.Compression.LZ4/LZ4Codec.cs#L139-L144

Note that there are 2 LZ4 libraries - one for streams (which this pr is using now) and one that doesn't use streams - https://www.nuget.org/packages/K4os.Compression.LZ4

Reply:
The two libraries actually implement different formats. The `K4os.Compression.LZ4.Streams` library implements the LZ4 frame format, and the `K4os.Compression.LZ4` library implements the LZ4 block format. Arrow IPC uses the frame format (https://github.com/apache/arrow/blob/apache-arrow-11.0.0/format/Message.fbs#L46-L48), so we need to use the streams library.

Reply:
I've opened https://github.com/MiloszKrajewski/K4os.Compression.LZ4/issues/79 to request this is added to the `K4os.Compression.LZ4` library.

Reply:
Thanks Eric, your issue prompted me to dig more into the `K4os` library though and I realised that since I initially started working on this, a new API has been added that allows using the frame format with more types than just `Stream`, so I've switched to this and removed the `CommunityToolkit.HighPerformance` dependency.